### PR TITLE
MAINT: Bessel tweaks

### PR DESCRIPTION
**Pull request by endolith**
I started making falling and rising factorial in another branch, but put a quick one here for now.

### CODE REVIEW

**Comment by argriffing** [comment]
the docstring backslashes are causing issues

**Comment by argriffing** [comment]
> I started making falling and rising factorial

scipy.special already has a rising factorial ufunc ([poch](http://docs.scipy.org/doc/scipy-0.17.0/reference/generated/scipy.special.poch.html)) but I guess it doesn't have an 'exact' keyword so it's not as useful for this PR, and ufuncs can't easily get keyword args.

**Comment by endolith** [comment]
Weird I didn't notice that.  This PR needs falling though, not rising.  (Also working on Thiran filters which use rising factorial.)   I also started a falling/rising factorial branch with `exact=` and broadcasting of inputs, but it's not finished yet.

**Comment by argriffing** [comment]
> This PR needs falling though, not rising.

FWIW if it weren't for the lack of `exact=` I think you could write falling_factorial(2*n - k, n) as poch(n - k + 1, n).

**Comment by person142** [comment]
There is a pretty standard procedure for adding keyword arguments to ufuncs in special--write a function for each value of the keyword, stick them in ``generate_ufuncs.py`` with underscored names, and then add an extra python wrapper which selects the right function. See e.g. the spherical Bessel functions (in 0.18, not the earlier ones) which have a keyword argument ``derivative``.

**Comment by endolith** [comment]
Anything else I need to do on this?

**Comment by Eric89GXL** [comment]
It looks reasonable to me, but can you briefly comment on what this accomplishes? Does it make it faster by not computing some redundant values that get cancelled out? Does it make it more accurate somehow?

File: scipy/signal/filter_design.py [hunk 1]
```diff
@@ -3095,8 +3096,8 @@ def ellipap(N, rp, rs):
 
     References
     ----------
-    Lutova, Tosic, and Evans, "Filter Design for Signal Processing", Chapters 5
-    and 12.
+    .. [1] Lutova, Tosic, and Evans, "Filter Design for Signal Processing",
```
Comment (endolith) [inline scipy/signal/filter_design.py]: I don't know what's in the reference, I was just fixing the formatting.

**Comment by endolith** [comment]
> Does it make it faster by not computing some redundant values that get cancelled out? 

Yes, it no longer computes (5*4*3*2*1)/(3*2*1) to get 5*4, for instance, and no longer computes all of the bessel polynomial when it only needs the last term.

> Does it make it more accurate somehow?

Yes, I also changed to calculate the magnitude normalization based on frequency response from poles instead of transfer functions so it's more accurate especially at higher orders.

**Comment by Eric89GXL** [comment]
Great, can you add a comment at least about the accuracy bit in the
relevant place(s) in the code if not already there?

**Comment by endolith** [comment]
Well it's in the commit message and I removed the previous comment about it being inaccurate at high orders.  Should I make the commit message more verbose?

**Comment by Eric89GXL** [comment]
The problem with only having it in a commit message (as opposed to in a
code comment) is that the next person looking at the code won't see it and
benefit from the expertise/rationale. But if you think it's clear enough
from just looking at the code then it's fine as is.

**Comment by endolith** [comment]
Ok I elaborated on the commit messages and rebased

**Comment by Eric89GXL** [comment]
Thanks @endolith

**Pull request merged**